src/msgs/updateClient.ts

```diff
@@ -21,19 +22,33 @@ export async function generateMsgUpdateClient(
   msg: MsgUpdateClient
   height: Height
 }> {
-  const latestHeight = Number(
+  const lastRevisionHeight = Number(
     ((await dstChain.rest.ibc.clientState(dstClientId)) as ClientState)
       .client_state.latest_height.revision_height
   )
-  const signedHeader = await getSignedHeader(srcChain)
+
+  let signedHeader = await getSignedHeader(srcChain)
+
   const header = signedHeader.header
   if (header === undefined) {
     throw Error('Header not found')
   }
   const currentHeight = Number(header.height)
   const validatorSet = await getValidatorSet(srcChain, currentHeight)
-  const trustedHeight = getRevisionHeight(latestHeight, header.chainId)
-  const trustedValidators = await getValidatorSet(srcChain, latestHeight + 1)
+  const trustedHeight = getRevisionHeight(lastRevisionHeight, header.chainId)
+  const trustedValidators = await getValidatorSet(
+    srcChain,
+    lastRevisionHeight + 1
+  )
+
+  // Keep querying until there is enough voting power
+  // Retry up to MAX_RETRY times in case the current height doesn't have sufficient voting power
+  const MAX_RETRY = 30
+  for (let i = 0; i < MAX_RETRY; i++) {
+    if (verifyVotingPower(signedHeader, validatorSet, trustedValidators)) break
+    signedHeader = await getSignedHeader(srcChain, currentHeight)
```

Comment:
is there a situation where we may need to retry on another height instead? something like maybe the public node the relayer using is catching up (please ignore if this cannot happen 🙏)